content/common/utils/utils.h

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#ifndef UTILS_H_
+#define UTILS_H_ 1
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef __FUNCTION_NAME__
+#ifdef WIN32  /* WINDOWS */
+	#define __FUNCTION_NAME__   __FUNCTION__  
+#else  /* *NIX */
+	#define __FUNCTION_NAME__   __func__ 
+#endif
+#endif
+
+#if defined(__linux__) || defined(__Unikraft__)
+#define print_err(call_desc)	perror(call_desc)
```

Comment:
I would actually give up the entire platform-specific string and only use `perror`. The `GetLastError` function is only useful for Win32 API calls. Otherwise, we would just use `strerror`. See [this discussion](https://stackoverflow.com/a/20057368/4804196). This will make everything generic.